src/main/scala/firrtl/Compiler.scala

```diff
@@ -576,6 +583,10 @@ trait Compiler extends Transform with DependencyAPIMigration {
     *   process by [[CompilerUtils.mergeTransforms]]
     * @return result of compilation
     */
+  @deprecated(
+    "Migrate to '(new FirrtlStage).execute(args: Array[String], annotations: AnnotationSeq)'." +
```

Comment:
Can you clarify what the arguments are to get the behavior of `compile`?

Reply:
I started to write this up, but it's complicated... the correct behavior is really going to be dependent on the child that extends `Compiler`. E.g., `VerilogCompiler` has different arguments than `HighFirrtlCompiler`. We can override the methods in the compiler children with more verbose deprecation warnings that show exactly how to invoke these. 

Alternatively, we can add a link to a gist that shows how to migrate (like with `DependencyAPIMigration`).

I don't personally think that there's huge benefit here as most people that are using the actual compilers are developers. The main API that I think people use is `ChiselStage`/`FirrtlStage`.

What do you think?

Reply:
I think the code that I would use to replace this is `new TransformManager(Seq(Dependency(Forms.LowForm), Dependency(CustomTransform)).execute(state)`. Would that be correct?

Reply:
For `LowFirrtlEmitter` that looks perfect.